internal/controller/progressive/progressive.go

```diff
@@ -245,6 +245,16 @@ func processUpgradingChild(
 	}
 	childStatus = rolloutObject.GetUpgradingChildStatus()
 
+	// check for Promote label here to force success logic
+	if rolloutObject.GetRolloutObjectMeta().Labels[common.LabelKeyNumaplanePromote] == "true" {
+		childStatus.ForcedSuccess = true
+		done, err := declareSuccess(ctx, rolloutObject, existingPromotedChildDef, existingUpgradingChildDef, childStatus, c)
+		if err != nil {
+			return done, false, 0, err
+		}
+		return done, false, assessmentSchedule.Interval, err
```

Comment:
for clarity, we may want to return 0 for the third parameter whenever `done` is true (same comment below), even though I realize that it's a little more code to do so